app/game/Swarm/App.hs

```diff
@@ -91,7 +90,7 @@ appMain opts = do
       eport <-
         Swarm.Web.startWebThread
           (userWebPort opts)
-          (mkReadonly appStateRef)
+          (readIORef appStateRef)
```

Comment:
Would be nice to add a comment here explaining that we are providing read-only access to the `IORef` by pre-applying `readIORef`, resulting in an `IO AppState` value which cannot do anything except provide the current value every time it is executed.